dashboard/legacy/middleware/files_api.rb

```diff
@@ -421,13 +415,17 @@ def put_file(endpoint, encrypted_channel_id, filename, body, project_type = nil)
       quota_crossed_half_used(endpoint, encrypted_channel_id) if quota_crossed_half_used?(app_size, body.length)
     end
 
-    # Block libraries with PII/profanity from being published.
-    # Block main.json file from aichat lab flagged with profanity from being saved.
-    #
-    # The "backpack" feature uses libraries to allow students to share code
-    # between their own projects -- skip this check for .java and .py files, since in this use case
-    # the files are only being used by a single user.
-    if endpoint == 'libraries' && BACKPACK_PROGRAM_FILE_TYPES.exclude?(file_type)
+    unless project_type
+      project = Projects.new(get_storage_id).get(encrypted_channel_id)
+      project_type = project[:projectType]&.downcase if project
+    end
+
+    # Block App Lab libraries with PII/profanity from being published and shared with other users.
+    # The "backpack" feature uses the libraries endpoint to allow users to share code
+    # between their own projects -- skip this check for backpack files since the files are
+    # only being used by a single user.
+    # Backpack is used in Java Lab, Python Lab, and Web Lab 2, but not in App Lab.
+    if endpoint == 'libraries' && project_type != 'backpack'
```

Comment:
FYI, `project_type` for `Backpack` assigned at:
https://github.com/code-dot-org/code-dot-org/blob/e0fd91e3fe943f93c65f91564707b897d6c126ab/dashboard/app/models/backpack.rb#L30